Microsoft.Toolkit.Uwp.UI/Triggers/EqualsStateTrigger.cs

```diff
@@ -0,0 +1,105 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Globalization;
+using Windows.UI.Xaml;
+
+namespace Microsoft.Toolkit.Uwp.UI.Triggers
+{
+    /// <summary>
+    /// Enables a state if the value is equal to another value
+    /// </summary>
+    /// <remarks>
+    /// <para>
+    /// Example: Trigger if a value is null
+    /// <code lang="xaml">
+    ///     &lt;triggers:EqualsStateTrigger Value="{Binding MyObject}" EqualTo="{x:Null}" />
+    /// </code>
+    /// </para>
+    /// </remarks>
+    public class EqualsStateTrigger : StateTriggerBase
+    {
+        private void UpdateTrigger() => SetActive(EqualsStateTrigger.AreValuesEqual(Value, EqualTo, true));
+
+        /// <summary>
+        /// Gets or sets the value for comparison.
+        /// </summary>
+        public object Value
+        {
+            get { return (object)GetValue(ValueProperty); }
+            set { SetValue(ValueProperty, value); }
+        }
+
+        /// <summary>
+        /// Identifies the <see cref="Value"/> DependencyProperty
+        /// </summary>
+        public static readonly DependencyProperty ValueProperty =
+            DependencyProperty.Register(nameof(Value), typeof(object), typeof(EqualsStateTrigger), new PropertyMetadata(null, OnValuePropertyChanged));
+
+        private static void OnValuePropertyChanged(DependencyObject d, DependencyPropertyChangedEventArgs e)
+        {
+            var obj = (EqualsStateTrigger)d;
+            obj.UpdateTrigger();
+        }
+
+        /// <summary>
+        /// Gets or sets the value to compare equality to.
+        /// </summary>
+        public object EqualTo
+        {
+            get { return (object)GetValue(EqualToProperty); }
+            set { SetValue(EqualToProperty, value); }
+        }
+
+        /// <summary>
+        /// Identifies the <see cref="EqualTo"/> DependencyProperty
+        /// </summary>
+        public static readonly DependencyProperty EqualToProperty =
+                    DependencyProperty.Register(nameof(EqualTo), typeof(object), typeof(EqualsStateTrigger), new PropertyMetadata(null, OnValuePropertyChanged));
+
+        internal static bool AreValuesEqual(object value1, object value2, bool convertType)
+        {
+            if (value1 == value2)
```

Comment:
Shouldn't it be value1.Equal(value2) with some additional null checks?
Since `((object)true == (object)true) -> false`

Reply:
That would throw if value1 is null though. What this is doing is doing a simple fast equals check, before going into a more advanced one below.